### DISCUSSION TITLE
Receiving WebSocket connect to 'wss://hostname:3035/ws' failed

### DISCUSSION THREAD

**Discussion by RailsCod3rFuture** [shakacode/shakapacker#54] [General]
During deployments to dev/production servers, everything works perfectly fine with websockets. Now, I am trying to use shakapacker on Docker and I receive "Unable to process ChannelName#action" Is there anything specific that I have to do, in order to communicate with web sockets using this version? I'm running Redis on the same container as the rails app and I am able to connect to it using 127.0.0.1. Every time that I use docker, I see [webpack-dev-server] Trying to reconnect.....? WebSocketClient @ WebSocketClient.js.

<img width="1379" alt="Screen Shot 2022-02-04 at 7 40 03 PM" src="https://user-images.githubusercontent.com/8632141/152621720-4b094a9d-e03e-4552-9856-01b67eb448b4.png">

**Comment by tomdracz** [comment]
If anything you might be facing Webpack rather than Shakapacker change. Default socket communication has changed from sockjs to native websocket (https://webpack.js.org/configuration/dev-server/#devserverwebsocketserver) - if you weren't using `ws` communication explicitly before, that might be a change for you. And by your error description, it does sound a little bit like you're hitting some existing ws connection set up for actioncable or similar 

You will need to play around with the dev server config and/or your docker setup. Try modifying it to use `sockjs` again or make necessary changes in your docker setup to accommodate for the ws sockets. Either way, it's not a Shakapacker issue.

Might not be matching what you have there but I had to change the following to get everything working smoothly with locals Rails app but connecting through dockerised nginx.

In dev server config in `webpacker.yml`
```
  dev_server:
    https: false
    host: '0.0.0.0'
    port: 3035
    client:
      webSocketURL:
        hostname: 'myhost.test'
        pathname: '/ws'
        port: 443
        protocol: 'wss'
 ...
```

In nginx config, new path added:

```
  location /ws {
    proxy_pass http://${DOCKER_HOST_IP}:${WEBPACKER_PORT};
    proxy_redirect off;
    proxy_http_version 1.1;
    proxy_set_header    Upgrade $http_upgrade;
    proxy_set_header    Connection "upgrade";
  }
  ```

**Comment by justin808** [comment]
I converted this to a discussion as this seems to be more of a webpack and not shakapacker issue.

**Comment by RailsCod3rFuture** [comment]
My belief is that our system might be too complex in this case -- given that we can't inject ENV variables into the webpacker.yml file....which pretty much tracks our hostname for subdomain related functionality.

**Reply by tomdracz** [reply]
@RailsCod3rFuture You can configure dev server with env variables though, might be helpful in your use case

https://github.com/shakacode/shakapacker/blob/master/README.md#development see more info here

**Reply by matoni109** [reply]
> 
@RailsCod3rFuture 
I have done this before by sending the env variables to `webpack` directly 

below was in my `Procfile.local`: 

`webpack: WHOAMI=$(whoami) bin/webpack-dev-server `

**Reply by RailsCod3rFuture** [reply]
Using this approach with the procfile in docker would actually be cool.
